### PR TITLE
Rework xml parser scan definitions

### PR DESCRIPTION
**Pull request by Honny1**
This PR changes:
- internal arrays to dictionaries for sreaching performance improvement
- loading negation for nodes

### CODE REVIEW

File: oval_graph/xml_parser.py [hunk 1]
```diff
@@ -109,20 +108,16 @@ def _get_oval_definitions(self):
...
-            if rule['id_rule'] == rule_id:
-                return rule['id_def']
-        raise ValueError('err- 404 rule not found!')
+        if rule_id in self.used_rules:
+            return dict(rule_id=rule_id,
+                        definition_id=self.used_rules[rule_id]['id_def'],
+                        definition=self.scan_definitions[self.used_rules[rule_id]['id_def']])
+        elif rule_id in self.notselected_rules:
+            raise ValueError(
+                'err- rule "{}" was not selected, so there are no results.'
```
Comment (matejak) [inline oval_graph/xml_parser.py]: Let's make this more user-friendly and representative (in another PR) - current error messages are quite geeky, and the client should not show the traceback, but it should only print the message.

File: oval_graph/_xml_parser_comments.py [hunk 2]
```diff
@@ -47,26 +46,25 @@ def _prepare_definition_comments(self):
...
-            definitions.append(comment_definition)
+            definitions[definition.get('id')] = comment_definition
         return definitions
 
     def _recursive_help_fill_comments(self, comments, nodes):
-        out = nodes
-        out['comment'] = comments['comment']
-        for node, comment in zip(out['node'], comments['node']):
+        for node, comment in zip(nodes, comments):
             node['comment'] = comment['comment']
```
Comment (matejak) [inline oval_graph/_xml_parser_comments.py]: Try to think about naming here - we have `node`, `comment`, `comment["node"]`, `node["node"]`, and even `comment["comment"]`, which is quite confusing. In graphs, you have nodes and edges (or relations), at the same time you have the data structure with comments, IDs, references etc., and you need to distinguish between them. Could you e.g. try to describe what is the node of a comment?

File: oval_graph/xml_parser.py (hunk 1 above)
Comment (Honny1) [reply oval_graph/xml_parser.py]: I create issue for this.

**Comment by matejak** [comment]
Much better, thanks!

**Pull request merged**